src/runtime/threading_backend.cc

```diff
@@ -167,7 +169,9 @@ class ThreadGroup::Impl {
       CPU_SET(id, &cpuset);
     }
 #if defined(__ANDROID__)
-    sched_setaffinity(thread, sizeof(cpu_set_t), &cpuset);
+    if (sched_setaffinity(pthread_gettid_np(thread), sizeof(cpu_set_t), &cpuset) != 0) {
```

Comment:
What if we will try to run this code on older versions than Android21+?

Reply:
If  on older versions than Android21+, there could be compile error with no func found.  In this case, could be record the tid  
like 
```cpp
std::mutex m;
std::map<std::thread::id, pid_t> threads;
void add_tid_mapping()
{
  std::lock_guard<std::mutex> l(m);
  threads[std::this_thread::get_id()] = syscall(SYS_gettid);
}
void wrap(void (*f)())
{
  add_tid_mapping();
  f();
}
```

Reply:
Just thinking how we can avoid such compilation error. Can we [check](https://developer.android.com/ndk/reference/group/apilevels) Android API version and for API>= 21 we will use `pthread_gettid_np` and in the opposite case - another implementation?

Reply:
I updated the with `__ANDROID_API__` macro, with api level >=21, use `pthread_gettid_np`, while < 21, get the thread tid due to the internal pthread implementation

Reply:
@echuraev